### PR TITLE
[W5.11][F12-A3]Zhang Ruiqi

### PR DESCRIPTION
**Pull request by zhangriqi**

### CODE REVIEW

**Comment by zhangriqi** [comment]
Work in progress...

**Comment by zhangriqi** [comment]
Enhancement: 1. Added a new command sort
2.Added four new classes: Block, Street, Unit, PostalCode
3.Updated the UserGuide and Developer Guid
4.Modified some tests, but didn't cover all

**Review by zhuleyan — COMMENTED** [review]
in Command.java, the newly added method getMessageForPersonSortShownSummary needs to be commented to show what does it do

**Review by wayneong95 — COMMENTED** [review]
Hello, correct me if i'm wrong but in your sortCommand.java, the description has not been updated to explain the function of the sort command. Also, i do not see where the sort command is actually implemented.

File: src/seedu/addressbook/commands/SortCommand.java [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    public static final String COMMAND_WORD = "sort";
+    public static final String MESSAGE_USAGE = COMMAND_WORD
+            + ": Displays all persons in the address book sorted by names.\n"
+            + "Example: " + COMMAND_WORD;
+     @Override
+     public CommandResult execute() {
+         List<ReadOnlyPerson> allPersons = addressBook.getAllPersons().immutableListView();
+         List<ReadOnlyPerson> sortedPersons = new ArrayList<ReadOnlyPerson>();
+         int i,j;
+         for(i=0;i<allPersons.size();i++){
```
Comment (OuyangDanwen) [inline src/seedu/addressbook/commands/SortCommand.java:22]: Try to use for each loop for better understanding.

File: src/seedu/addressbook/commands/SortCommand.java [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+             sortedPersons.add(allPersons.get(i));
+             }
+         int n=allPersons.size();
+         for(i=0;i<n-1;i++){
```
Comment (OuyangDanwen) [inline src/seedu/addressbook/commands/SortCommand.java:26]: Is this the best way to sort the list. Can you think of a better way to do this, eg. use of Java in built comparator etc. Refer to your peers' code if you are not sure.

**Review by OuyangDanwen — COMMENTED** [review]
Generally okay. Take note of the following: 1. Please submit a clean PR for reviewing, i.e. branch properly and do not stack commits for various exercises in one PR. 2. Try to be more familiar with Java and Java coding standard. You can close the PR after reading the comments. 3. Commit message should be in imperative form, eg. start with a verb in present tense. You should have one more feature for 2kLoC as you missed 1kLoC. You can do so by this week's tutorial. Do this in a separate clean PR to claim the task fully.

File: src/seedu/addressbook/commands/SortCommand.java (hunk 2 above)
Comment (zhangriqi) [reply src/seedu/addressbook/commands/SortCommand.java]: Do I need to do all these enhancements in another PR? Or only one feature in the new PR will do?

Comment (zhangriqi) [reply src/seedu/addressbook/commands/SortCommand.java:26]: Thank you

**Pull request closed without merging**